sqllogictest/src/parser.rs

```diff
@@ -102,8 +101,7 @@ pub enum Record<T: ColumnType> {
         label: Option<String>,
         /// The SQL command is expected to fail with an error messages that matches the given
         /// regex. If the regex is an empty string, any error message is accepted.
-        #[educe(PartialEq(method = "cmp_regex"))]
-        expected_error: Option<Regex>,
+        expected_error: Option<ExpectedError>,
```

Comment:
First I'm thinking whether it's necessary to introduce the error enum. Why not put it in `expected_results`. But it should be problematic.

I think we've messed the `Record` type. 🤣 To be more correct, it should be like enum of error or results. Some fields are common, and some belong only to query.

Reply:
nvm. I'm happy to live with it.

Reply:
Same opinion with you. 😕 It's a sum type instead of a product type. We may refactor them in future PRs.